File: Week 07/id_454/LeetCode_56_454.go
func merge(intervals []Interval) []Interval {
    if len(intervals) == 0 {
        return []Interval{}
    }
    
    sort.Slice(intervals, func(i, j int) bool {return intervals[i].Start < intervals[j].Start})
    ret := []Interval{intervals[0]}
    for i := 1; i < len(intervals); i++ {
        tail := ret[len(ret)-1]
        if tail.End < intervals[i].Start {
            ret = append(ret, intervals[i])
        } else if intervals[i].End > tail.End {
            ret[len(ret)-1].End = intervals[i].End
        }
    }
    return ret
}

File: Week 07/id_454/LeetCode_56_454.php
<?php
class Solution
{
    function merge($intervals)
    {
        asort($intervals);
        $range = array();
        foreach ($intervals as $interval) {
            if (empty($range)) {
                array_push($range, $interval);
                continue;
            }
            $inRange = false;
            foreach ($range as $index => $value) {
                if ($value[0] <= $interval[0] && $value[1] >= $interval[1]) {
                    $inRange = true;
                    continue;
                }
                if ($value[0] <= $interval[0] && $value[1] < $interval[1] && $interval[0] <= $value[1]) {
                    $range[$index][1] = $interval[1];
                    $inRange = true;
                }
                if ($value[0] > $interval[0] && $value[1] >= $interval[1] && $value[1] >= $interval[0]) {
                    $range[$index][0] = $interval[0];
                    $inRange = true;
                }
            }
            if (!$inRange) {
                array_push($range, $interval);
            }
        }
        return $range;
    }
}
